src/getrepos.ts

```diff
@@ -22,9 +21,9 @@ export async function getRepoIds(repoPath: string, clientInput: {account, access
     }
     const newRepoPath = resolve(repoPath, repo);
     const newRepos = readdirSync(newRepoPath);
-    subdir = false;
+    let subdir = false;
 
-    if (!(newRepos.includes('package.json')) && await containsSubdirectory(newRepos, j1Client)) {
+    if (!(newRepos.includes('package.json')) && await containsSubdirectory(newRepos, j1Client, subdir)) {
```

Comment:
Ah, I should have clarified what I meant when I said I didn't like the top-level subdir variable. It's a side effect when functions mutate global state or the state passed to them. Such behavior often causes subtle bugs as well as being generally bad practice. Can subdir just become a local variable in `containsSubdirectory`, since it's only read once and that read is in `containsSubdirectory`?

Reply:
Actually a few lines up on 24 I reset the value to false for the next thing being checked.

Reply:
Oh wait I think I see what you mean actually